pkg/controller/common/rollout/rollout_plan_controller.go

```diff
@@ -357,6 +359,15 @@ func (r *Controller) GetWorkloadController() (workloads.WorkloadController, erro
 			return workloads.NewDeploymentScaleController(r.client, r.recorder, r.parentController,
 				r.rolloutSpec, r.rolloutStatus, target), nil
 		}
+
+		// check if the target workload is StatefulSet
+		if r.targetWorkload.GetKind() == reflect.TypeOf(apps.StatefulSet{}).Name() {
+			if r.sourceWorkload == nil {
+				return workloads.NewStatefulSetScaleController(r.client, r.recorder, r.parentController,
+					r.rolloutSpec, r.rolloutStatus, target), nil
+			}
+		}
 	}
```

Comment:
Please delete. When next pr (`statefulset rollout controller` )ready we can add this back.